devito/symbolics/inspection.py

```diff
@@ -20,48 +20,60 @@ def count(exprs, query):
     return dict(mapper)
 
 
-def estimate_cost(expr, estimate_functions=False):
+def estimate_cost(exprs, estimate=False):
     """
     Estimate the operation count of an expression.
 
+
     Parameters
     ----------
-    expr : expr-like or list of expr-like
+    exprs : expr-like or list of expr-like
         One or more expressions for which the operation count is calculated.
-    estimate_functions : dict, optional
-        A mapper from known functions (e.g., sin, cos) to (estimated) operation counts.
+    estimate : bool, optional
+        Defaults to False; if True, the following rules are applied:
+            * Trascendental functions (e.g., cos, sin, ...) count as 50 ops.
+            * Divisions (powers with a negative exponened) count as 25 ops.
     """
-    external_functions = {sin: 50, cos: 50}
+    trascendentals_cost = {sin: 50, cos: 50, exp: 50, log: 50}
+    div_cost = 25
+
     try:
-        # Is it a plain SymPy object ?
-        iter(expr)
-    except TypeError:
-        expr = [expr]
+        # Is it a plain symbol/array ?
+        if exprs.is_AbstractFunction or exprs.is_AbstractSymbol:
+            return 0
+    except AttributeError:
+        pass
     try:
         # Is it a dict ?
-        expr = expr.values()
+        exprs = exprs.values()
     except AttributeError:
         try:
-            # Must be a list of dicts then
-            expr = flatten([i.values() for i in expr])
-        except AttributeError:
+            # Could still be a list of dicts
+            exprs = flatten([i.values() for i in exprs])
+        except (AttributeError, TypeError):
             pass
     try:
         # At this point it must be a list of SymPy objects
         # We don't use SymPy's count_ops because we do not count integer arithmetic
         # (e.g., array index functions such as i+1 in A[i+1])
         # Also, the routine below is *much* faster than count_ops
-        expr = [i.rhs if i.is_Equality else i for i in expr]
-        operations = flatten(retrieve_ops(i) for i in expr)
+        exprs = [i.rhs if i.is_Equality else i for i in as_tuple(exprs)]
+        operations = flatten(retrieve_xops(i) for i in exprs)
         flops = 0
         for op in operations:
             if op.is_Function:
-                if estimate_functions:
-                    flops += external_functions.get(op.__class__, 1)
+                if estimate:
+                    flops += trascendentals_cost.get(op.__class__, 1)
+                else:
+                    flops += 1
+            elif op.is_Pow:
```

Comment:
This is unlikely to crop up in practice, but I didn't think powers were cheap at all.  Especially a**b where a and b are floats.  If b is an int, then cost is ~(b-1)?